zerver/tests/test_signup.py

```diff
@@ -572,6 +572,37 @@ def test_login_bad_password_rate_limiter(self) -> None:
 
         remove_ratelimit_rule(10, 2, domain='authenticate_by_username')
 
+    def test_email_auth_weak_password(self) -> None:
+        user_profile = self.example_user('hamlet')
+        weak_password = "weak_password_of_25_chars"
+        strong_password = "a_very_strong_password_of_34_chars"
+        user_profile.set_password(weak_password)
+        user_profile.save()
+        with self.settings(PASSWORD_MIN_LENGTH=30):
+
+            # First check if user is taken to reset page.
+            data = dict(username=user_profile.delivery_email,
+                        password=weak_password)
+            result = self.client_post('/accounts/login/', data)
+            result = self.login_with_return(user_profile.delivery_email, weak_password)
+            self.assertEqual(result.status_code, 302)
+            result = self.client_get(result.url)
+            self.assertEqual(result.status_code, 302)
+            self.assertTrue((result.url).startswith('http://zulip.testserver/accounts/weak_password/reset'))
+            result = self.client_get(result.url)
+            self.assertEqual(result.url, '/accounts/weak_password/reset/MTA/set-password/')
+            result = self.client_get(result.url)
+            self.assert_in_success_response(['Weak credentials. Please Update your password to continue using Zulip.'],
+                                            result)
+
+            # Now change password and verify user isn't redirected to reset page.
+            user_profile.set_password(strong_password)
+            user_profile.save()
+            self.logout()
+            result = self.login_with_return(user_profile.delivery_email, strong_password)
+            result = self.client_get(result.url)
+            self.assertNotEqual(result.status_code, 302)
```

Comment:
I'm not sure if this is right way to check if the user isn't getting redirected to reset page. Could you confirm @mateuszmandera ?

Reply:
Let's assert what the status_code should be, rather than what it isn't. If the server returned `500` here, the test would pass in its current form, but we wouldn't be happy with that status code. Let's also assert what the url is (`/` I think?)